app/scripts/components/common/map/style-generators/hooks.ts

```diff
@@ -106,8 +204,187 @@ export function useCMR({ id, stacCol, stacApiEndpointToUse, date, assetUrlReplac
     return () => {
       controller.abort();
     };
-  }, [id, stacCol, stacApiEndpointToUse, date, assetUrlReplacements, stacApiEndpoint, onStatusChange]);
+  }, [id, stacCol, tileApiEndpointToUse, stacApiEndpointToUse, date, onStatusChange, sourceParams]);
+
+  return tileUrls;
+};
+
+export function useMosaic(options: UseDatasetTilesOptions) {
```

Comment:
Nice! 👏🏼 Great idea on moving `mosaic` and `stacCollection` logic out of raster-timeseries file and into their own hooks here to be reused. Also cleans up that file a ton...